pkg/splunk/enterprise/configuration.go

```diff
@@ -308,11 +306,41 @@ func validateCommonSplunkSpec(spec *enterpriseApi.CommonSplunkSpec) error {
 		return fmt.Errorf("Negative value (%d) is not allowed for Readiness probe intial delay", spec.ReadinessInitialDelaySeconds)
 	}
 
+	// if not provided, set default values for imagePullSecrets
+	err := ValidateImagePullSecrets(ctx, c, cr, spec)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think we can conditionally call, if the slice is not empty?

Reply:
In order to retain the logic [here](https://github.com/splunk/splunk-operator/pull/730/files#diff-57923b610c6cabf2ed47c4c3b44b4b365366b4949242100ff5c9856e02562424R550), we need to assign `spec.ImagePullSecrets` a nil slice? I am not sure if the CR spec automatically assigns the nil slice in which case we don't need to assign(I have seen issues before where we ran into issues for not assigning values to all fields of CR spec? i.e in integration tests).

Reply:
Ideally, the spec should have been tagged as `omitempty` which will be ignored by the API server. And I think this is important IMO. As all our docker images are public, the customer may not even configure the secrets, instead, he can just set up open read permissions for the repository. So, we should make this config optional. thoughts?

Reply:
Currently, the config for `imagePullSecrets` from the CR spec is optional. But the statefulSet is always populated i.e even if we have no CR spec a nil slice is updated. Let's say there is no CR spec and you point it to a public repo image, that image from is pulled cleanly. I agree that we should only warn the user upon not finding any of the secrets in the config!